### PR TITLE
Resolve #1253, #1331

### LINKED ISSUE #1253: save btn should be disabled before checkWalletConnection finished

### PR DESCRIPTION
**Pull request by InVincible2016**

### CODE REVIEW

File: src/tests/e2e/emailTransferFormEntryPoint.test.e2e.js [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      const recipients = await recipientPage.getRecipientList()
+      let recipient = {}
+      // assume there is at least one recipient
+      if (recipients.length >= 1) {
+        recipient = recipients[0]
+        await page.click(`[data-test-id="send_to_recipient_${recipient.email}"]`)
+      }
+
+      const emtPage = new EmailTransferFormPage()
+      //  field should be empty
```
Comment (timothywangdev) [inline src/tests/e2e/emailTransferFormEntryPoint.test.e2e.js]: comment seems misleading

File: src/wallet.js [hunk 2]
```diff
@@ -80,6 +80,7 @@ function isSendable ({ status, walletType }: { status: Object, walletType: strin
       'coinbaseWalletLink'
     ].includes(walletType)
   ) {
+    console.log('isMobile', status.sendable)
```
Comment (timothywangdev) [inline src/wallet.js]: remove

File: src/tests/e2e/pages/emailTransferForm.page.js [hunk 3]
```diff
@@ -0,0 +1,125 @@
+import log from 'loglevel'
+import { getWalletTitle } from '../../../wallet.js'
```
Comment (timothywangdev) [inline src/tests/e2e/pages/emailTransferForm.page.js]: don't we need to add moduleNameMapper in jest.config.js to make this work?

File: src/tests/e2e/pages/accounts.page.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
+import log from 'loglevel'
+log.setDefaultLevel('info')
+
+class AccountsManagementPage {
+  async getAccountsList () {
+    const accountsElementList = await page.$$('[data-test-id^="account_list_item"]')
```
Comment (timothywangdev) [inline src/tests/e2e/pages/accounts.page.js]: is "^" intentional?

File: src/tests/e2e/pages/accounts.page.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+        const walletPlatformHandle = await elementHandle.$('[data-test-id="wallet_platform"]')
+        const assetsHandle = await elementHandle.$('[data-test-id="assets_cell"]')
+        const walletType = (
+          await (await walletPlatformHandle.getProperty('textContent')).jsonValue()
+        ).split(', ')[0]
+        const platformType = (
+          await (await walletPlatformHandle.getProperty('textContent')).jsonValue()
+        ).split(', ')[1]
+        let assets = await (await assetsHandle.getProperty('textContent')).jsonValue()
+        while (!assets) {
```
Comment (timothywangdev) [inline src/tests/e2e/pages/accounts.page.js]: is it possible to simply wait till skeleton does not show up anymore

Comment (InVincible2016) [reply src/tests/e2e/pages/accounts.page.js]: It's basically the same thing

File: src/tests/e2e/pages/accounts.page.js (hunk 4 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/accounts.page.js]: yes, select all data-test-id start with account_list_item

File: src/tests/e2e/pages/emailTransferForm.page.js (hunk 3 above)
Comment (InVincible2016) [reply src/tests/e2e/pages/emailTransferForm.page.js]: That part has been merged.

File: src/tests/e2e/pages/accounts.page.js (hunk 5 above)
Comment (timothywangdev) [reply src/tests/e2e/pages/accounts.page.js]: we can do something like this to better control the timeout. Currently approach depends on test timeout
await page.waitForSelector('div', {hidden: true, timeout: xxx});

Comment (InVincible2016) [reply src/tests/e2e/pages/accounts.page.js]: but we dont know how long it might take

Comment (timothywangdev) [reply src/tests/e2e/pages/accounts.page.js]: good point

**Pull request merged**